src/montgomerybuilder.js

```diff
@@ -311,6 +313,154 @@ function buildFromMontgomery(fn, q) {
     });
 }
 
+// no adx assembler
+
+function templateMontgomery_no_adx(fn, q, round) {
+
+    const n64 = Math.floor((q.bitLength() - 1) / 64)+1;
+    const canOptimizeConsensys = q.shiftRight((n64-1)*64).leq( bigInt.one.shiftLeft(64).minus(1).shiftRight(1).minus(1) );
+    const c = new AsmBuilder(fn, 4 + n64 + 1 + (canOptimizeConsensys ? 0 : 1));
+    const t = 4;
+
+    const params = {q, n64, t, canOptimizeConsensys};
+
+
+    let r0, r1, r2;
+    function setR(step) {
+        if ((step % 3) == 0) {
+            r0 = "r8";
+            r1 = "r9";
+            r2 = "r10";
+        } else if ((step % 3) == 1) {
+            r0 = "r9";
+            r1 = "r10";
+            r2 = "r8";
+        } else {
+            r0 = "r10";
+            r1 = "r8";
+            r2 = "r9";
+        }
+    }
+
+    c.op("mov", "rbp", "rsp")
+    c.op("sub","rsp", "" + n64 * 8);
+    c.op("mov","rcx","rdx");
+    c.op("mov","r11", "[ np ]");
+    c.op("xor","r8","r8");
+    c.op("xor","r9","r9");
+    c.op("xor","r10","r10");
+    // Main loop
```

Comment:
```suggestion

    // Main loop
```
How about adding an empty line here?

Reply:
done